compiler/modules/front/stmt.bal

```diff
@@ -375,14 +380,13 @@ function invalidateAssignments(Assignment[] assignments, Narrowing[] narrowings)
 }
 
 function unusedLocalVariables(StmtContext cx, Environment env, Binding? bindingLimit) returns CodeGenError? {
```

Comment:
I wonder about using a Binding to represent bindingLimit. You are relying on each Binding having a distinct (!==) identity. Would it be better to use a BindingChain here?

Reply:
Do you mean to use the distinct identity of `BindingChain` instead of `Binding`? I'll try that.

Another thing that makes Bindings confusing is `()` is used with two different meanings. In `Environment`, bindings being nil indicates the end of the BindingChain. In `StmtEffect`, bindings being nil indicates statement didn't produce any Bindings.

I'll try below

```ballerina
const EMPTY_CHAIN = false;

type BindingChain EMPTY_CHAIN|record {|
    Binding head;
    BindingChain prev;
|};
```